src/main/java/Deadline.java

```diff
@@ -0,0 +1,19 @@
+public class Deadline extends Task{
+    private String ddl;
+
+    public Deadline(String input) {
+        //"project meeting /from Mon 2pm /to 4pm"
+        super(input.split("/")[0].strip());
+        String[] tokens = input.split("/");
+        //"from Mon 2PM"
+        this.ddl = tokens[1].split("by")[1].strip();
+    }
+
+    @Override
+    public String toString() {
+        if (super.done){
+            return "[D][X] " + super.taskContent + String.format(" (by: %s)", this.ddl);
+        }
+        return "[D][ ] " + super.taskContent + String.format(" (by: %s)", this.ddl);
```

Comment:
Consider retaining the function in Week 2 that the Prof provided for returning if the status is completed. 

From week 2 / Level-3 / `Task.java`
```java
    public String getStatusIcon() {
        return (isDone ? "X" : " "); // mark done task with X
    }
```